recipes/darwin-toolchain/all/conanfile.py

```diff
@@ -0,0 +1,117 @@
+    def config_options(self):
+        # build_type is only useful for bitcode
+        if self.settings.os == "Macos":
+            del self.settings.build_type
+            del self.options.bitcode
+
+    def validate(self):
+        if not tools.is_apple_os(self.settings.os):
+            raise ConanInvalidConfiguration("os must be an Apple os")
+        if self.settings.os in ["watchOS", "tvOS"] and not self.options.bitcode:
+            raise ConanInvalidConfiguration("bitcode is required on watchOS/tvOS")
+        if self.settings.os == "watchOS" and self.settings.arch not in ["armv7k", "armv8", "x86", "x86_64"]:
+            raise ConanInvalidConfiguration("watchOS: Only supported archs: [armv7k, armv8, x86, x86_64]")
+
+    def package(self):
+        self.copy("darwin-toolchain.cmake")
+
+    def package_info(self):
+        darwin_arch = tools.to_apple_arch(self.settings.arch)
+
+        if self.settings.os == "watchOS" and self.settings.arch == "armv8":
+            darwin_arch = "arm64_32"
+
+        sdk = None
+        if self.settings.os == "iOS":
+            if self.settings.get_safe("os.sdk") == "iPhoneSimulator":
+                sdk = "iphonesimulator"
+            else:
+                sdk = "iphoneos"
+        xcrun = tools.XCRun(self.settings, sdk)
+        sysroot = xcrun.sdk_path
+
+        self.cpp_info.sysroot = sysroot
+
+        common_flags = ["-isysroot%s" % sysroot]
+
+        if self.settings.get_safe("os.version"):
+            common_flags.append(tools.apple_deployment_target_flag(self.settings.os, self.settings.os.version, os_sdk=sdk))
+
+        if not self.settings.os == "Macos" and self.options.bitcode:
+            if self.settings.build_type == "Debug":
+                bitcode_flag = "-fembed-bitcode-marker"
+            else:
+                bitcode_flag = "-fembed-bitcode"
+            common_flags.append(bitcode_flag)
+
+        # CMake issue, for details look https://github.com/conan-io/conan/issues/2378
+        cflags = copy.copy(common_flags)
+        cflags.extend(["-arch", darwin_arch])
+        self.cpp_info.cflags = cflags
+        self.cpp_info.cxxflags = cflags
+        link_flags = copy.copy(common_flags)
+        link_flags.append("-arch %s" % darwin_arch)
+
+        self.cpp_info.sharedlinkflags.extend(link_flags)
+        self.cpp_info.exelinkflags.extend(link_flags)
+
+        # Set flags in environment too, so that CMake Helper finds them
+        cflags_str = " ".join(cflags)
+        ldflags_str = " ".join(link_flags)
+        self.env_info.CC = xcrun.cc
+        self.env_info.CPP = "%s -E" % xcrun.cc
+        self.env_info.CXX = xcrun.cxx
+        self.env_info.AR = xcrun.ar
+        self.env_info.RANLIB = xcrun.ranlib
+        self.env_info.STRIP = xcrun.strip
+
+        self.env_info.CFLAGS = cflags_str
+        self.env_info.ASFLAGS = cflags_str
+        self.env_info.CPPFLAGS = cflags_str
+        self.env_info.CXXFLAGS = cflags_str
+        self.env_info.LDFLAGS = ldflags_str
```

Comment:
I'm an Apple newb, so apologies for this question.
What are the actual values used for `CC/CPP/CXX/AR/RANLIB/STRIP`?
Do they contain build system specific paths?

I was just wonder if it would be possible to, instead of using environment variables,
to create *custom* cmake toolchain files in `package` with the values embedded in the toolchain file.

Reply:
>What are the actual values used for CC/CPP/CXX/AR/RANLIB/STRIP?

It's whatever path `xcrun` returns, which is located inside Xcode internals, so there is no absolute hardcoded path in the recipe.

>I was just wondering if it would be possible to, instead of using environment variables,
to create custom cmake toolchain files in package with the values embedded in the toolchain file.

Answered a similar comment here https://github.com/conan-io/conan-center-index/pull/5545#discussion_r634143051